starter-code/src/contact.js

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from "react";
+import './App.css';
+import contacts from './contacts.json';
+
+class Contact extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            contact: contacts.sort(() => Math.random() - 0.5),
+            clickCount: 5
+        }
+    }
+    clickHandler = () => {
+        this.setState({
+            clickCount: this.state.clickCount + 1
+        })
+    }
+    sortByName = () => {
+        this.setState({
+            contact: contacts.sort((a, b) => a.name > b.name)//Ne pas mettre de slice à cet endroit, sinon on bloque l'array à une certaine taille
+        })
+    }
+
```

Comment:
When sorting alphabetically you have to return a negative number to keep the same order and a positive one to invert it (see example below). Also, don't forget to uppercase or lowercase, in case two elements start by the same letter and sort needs to look at the second item:
```js
// sort by name
items.sort(function(a, b) {
  var nameA = a.name.toUpperCase(); // ignore upper and lowercase
  var nameB = b.name.toUpperCase(); // ignore upper and lowercase
  if (nameA < nameB) {
    return -1;
  }
  if (nameA > nameB) {
    return 1;
  }

  // names must be equal
  return 0;
});
```